lua/hop/hint.lua

```diff
@@ -108,19 +140,32 @@ function M.mark_hints_line(hint_mode, line_nr, line, col_offset, win_width)
   local shifted_line = line:sub(1 + col_offset, end_index)
 
   local col = 1
-  while true do
+
+  local line_too_late = hint_mode.direction == AFTER_CURSOR and line_nr + 1 < cursor_pos[1]
+  local line_too_early = hint_mode.direction == BEFORE_CURSOR and line_nr + 1 > cursor_pos[1]
+  local line_not_exact = line_nr + 1 ~= cursor_pos[1] and hint_mode.same_line
+  local linewise_excludes_current = line_nr + 1 == cursor_pos[1] and hint_mode.linewise
+  local is_invalid = line_too_early or line_too_late or line_not_exact or linewise_excludes_current
```

Comment:
I don’t think the logic of the line should be handled here, since this function is going to be called for a line we know we want to hint.

Also, can you add comments explaining what these bools are for?